### PR TITLE
feat: customPath changes

### PR DESCRIPTION
**Pull request by krpeacock**
# Description

feat: new CustomPath class, better docs, and deprecating metadata path type for CanisterStatus

# Checklist:

- [x] My changes follow the guidelines in [CONTRIBUTING.md](https://github.com/dfinity/agent-js/blob/main/CONTRIBUTING.md).
- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I have edited the CHANGELOG accordingly.
- [x] I have made corresponding changes to the documentation.

### CODE REVIEW

File: packages/agent/src/canisterStatus/index.ts [hunk 1]
```diff
@@ -229,7 +245,12 @@ export const request = async (options: {
   });
 
   // Fetch all values separately, as each option can fail
-  await Promise.all(promises);
+  try {
```
Comment (ericswanson-dfinity) [inline packages/agent/src/canisterStatus/index.ts:248]: leftover from testing?

File: packages/agent/src/canisterStatus/index.ts [hunk 2]
```diff
@@ -211,6 +225,8 @@ export const request = async (options: {
           }
         }
       } catch (error) {
+        error;
+        console.log(error);
```
Comment (ericswanson-dfinity) [inline packages/agent/src/canisterStatus/index.ts:229]: leftover from testing?

File: packages/agent/src/canisterStatus/index.ts [hunk 3]
```diff
@@ -229,7 +245,12 @@ export const request = async (options: {
   });
 
   // Fetch all values separately, as each option can fail
-  await Promise.all(promises);
+  try {
+    await promises[0];
```
Comment (dfx-json) [inline packages/agent/src/canisterStatus/index.ts:249]: should this go in a `for (const promise of promises)...` if you want to fetch each value separately? alternatively, theres `Promise.allSettled`

File: packages/agent/src/canisterStatus/index.ts [hunk 4]
```diff
@@ -94,7 +107,7 @@ export type CanisterStatusOptions = {
 };
 
 /**
- *
+ * Request information in the
```
Comment (ericswanson-dfinity) [inline packages/agent/src/canisterStatus/index.ts:110]: in the what?

**Review by dfx-json — APPROVED** [review]

File: packages/agent/src/canisterStatus/index.ts [hunk 5]
```diff
@@ -128,8 +141,9 @@ export const request = async (options: {
     return (async () => {
       try {
         const response = await agent.readState(canisterId, {
-          paths: [encodedPaths[index]],
+          paths: encodedPaths,
         });
+        response; //??
```
Comment (ericswanson-dfinity) [inline packages/agent/src/canisterStatus/index.ts:146]: leftover from testing?

File: packages/agent/src/canisterStatus/index.ts (hunk 2 above)
Comment (krpeacock) [reply packages/agent/src/canisterStatus/index.ts:229]: Oh wait, this had changes from another feature left in it, one sec

File: packages/agent/src/canisterStatus/index.ts [hunk 6]
```diff
@@ -128,8 +141,9 @@ export const request = async (options: {
     return (async () => {
       try {
         const response = await agent.readState(canisterId, {
-          paths: [encodedPaths[index]],
+          paths: encodedPaths,
```
Comment (ericswanson-dfinity) [inline packages/agent/src/canisterStatus/index.ts:144]: does this change some behavior that should be called out in the changelog?

**Review by ericswanson-dfinity — APPROVED** [review]

**Comment by krpeacock** [comment]
Sorry for the chaotic PR - there were some exploratory changes left over from `SDK-1219` when I thought this file was only modified from the CustomPath portion

**Pull request merged**